calicoctl/resourcemgr/resourcemgr.go

```diff
@@ -198,16 +198,28 @@ func (rh resourceHelper) Update(ctx context.Context, client client.Interface, re
 	// If the resourceVersion is specified then we use it to try and update the resource.
 	// Do not attempt to retry if the resource version is specified.
 	if rv != "" {
+		// Clean out the resource version to always get the latest revision.
+		ru := resource.DeepCopyObject().(ResourceObject)
```

Comment:
It looks as though mergeMetadataForUpdate() adjusts the resource that you give it, so I think the branch of code below this if `if rv != ""` conditional will also modify the input resource.

Therefore, probably best just to move this DeepCopy to outside this conditional.  Maybe just have `resource = resource.DeepCopyObject().(ResourceObject)` so it's not possible to accidentally modify the original resource.

Reply:
Good point. Definitely better to make it safer.